src/screens/points/claim-flow/ClaimRewardsPanel.tsx

```diff
@@ -179,24 +190,99 @@ const ClaimingRewards = ({
+      const opEth = {
+        ...opEth_,
+        chainName: chainNameFromChainId(ChainId.optimism),
+      };
+
+      // fetch the native asset from the destination chain
+      let destinationEth_;
+      if (chainId === ChainId.base) {
+        destinationEth_ = await ethereumUtils.getNativeAssetForNetwork(getNetworkFromChainId(ChainId.base));
+      } else if (chainId === ChainId.zora) {
+        destinationEth_ = await ethereumUtils.getNativeAssetForNetwork(getNetworkFromChainId(ChainId.zora));
+      } else {
+        destinationEth_ = opEth;
+      }
+
+      // Add missing properties to match types
+      const destinationEth = {
+        ...destinationEth_,
+        chainName: chainNameFromChainId(chainId as ChainId),
+      };
+
+      const selectedGas = {
+        maxBaseFee: meteorologyData?.fast.maxBaseFee,
+        maxPriorityFee: meteorologyData?.fast.maxPriorityFee,
+      };
 
-      if (claimInfo?.error) {
-        setClaimStatus(claimInfo?.error.type);
+      let gasParams: TransactionGasParamAmounts | LegacyTransactionGasParamAmounts = {} as
+        | TransactionGasParamAmounts
+        | LegacyTransactionGasParamAmounts;
+
+      gasParams = {
+        maxFeePerGas: selectedGas?.maxBaseFee as string,
+        maxPriorityFeePerGas: selectedGas?.maxPriorityFee as string,
+      };
+      const gasFeeParamsBySpeed = getGasSettingsBySpeed(ChainId.optimism);
+
+      const actionParams = {
+        address,
+        toChainId: chainId,
+        sellAmount: claimable as string,
+        chainId: ChainId.optimism,
+        assetToSell: opEth as ParsedAsset,
+        assetToBuy: destinationEth as ParsedAsset,
+        quote: undefined,
+        // @ts-expect-error - collision between old gas types and new
+        gasFeeParamsBySpeed,
+        gasParams,
+      } satisfies RapSwapActionParameters<'claimBridge'>;
+
+      const provider = await getProviderForNetwork(Network.optimism);
+      const wallet = await loadWallet(address, false, provider);
+      if (!wallet) {
+        Alert.alert(i18n.t(i18n.l.swap.unable_to_load_wallet));
+        return { nonce: null };
       }
 
-      // clear and refresh claim data so available claim UI disappears
-      invalidatePointsQuery(address);
-      await refetch();
+      try {
+        const { errorMessage, nonce: bridgeNonce } = await walletExecuteRap(
+          wallet,
+          'claimBridge',
+          // @ts-expect-error - collision between old gas types and new
+          actionParams
+        );
+
+        if (errorMessage) {
+          setClaimStatus('error');
+          return { nonce: null };
+        }
 
-      return claimInfo;
+        if (typeof bridgeNonce === 'number' && bridgeNonce >= 0) {
+          // clear and refresh claim data so available claim UI disappears
+          invalidatePointsQuery(address);
+          refetch();
+          return { nonce: bridgeNonce };
+        } else {
+          setClaimStatus('error');
```

Comment:
@christianbaroni  I've added this new state `error` so you can show something specific if bridging fails